homeassistant/components/overkiz/climate_entities/atlantic_electrical_heater_with_adjustable_temperature_setpoint.py

```diff
@@ -0,0 +1,137 @@
+"""Support for Atlantic Electrical Heater (With Adjustable Temperature Setpoint)."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.climate import (
+    PRESET_BOOST,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+PRESET_AUTO = "auto"
+PRESET_COMFORT1 = "comfort-1"
+PRESET_COMFORT2 = "comfort-2"
+PRESET_FROST_PROTECTION = "frost_protection"
+PRESET_PROG = "prog"
+
+
+# Map Overkiz presets to Home Assistant presets
+OVERKIZ_TO_PRESET_MODE = {
+    OverkizCommandParam.OFF: PRESET_NONE,
+    OverkizCommandParam.FROSTPROTECTION: PRESET_FROST_PROTECTION,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.COMFORT_1: PRESET_COMFORT1,
+    OverkizCommandParam.COMFORT_2: PRESET_COMFORT2,
+    OverkizCommandParam.AUTO: PRESET_AUTO,
+    OverkizCommandParam.BOOST: PRESET_BOOST,
+    OverkizCommandParam.INTERNAL: PRESET_PROG,
+}
+
+PRESET_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODE.items()}
+
+# Map Overkiz HVAC modes to Home Assistant HVAC modes
+OVERKIZ_TO_HVAC_MODE = {
+    OverkizCommandParam.ON: HVACMode.HEAT,
+    OverkizCommandParam.OFF: HVACMode.OFF,
+    OverkizCommandParam.AUTO: HVACMode.AUTO,
+    OverkizCommandParam.BASIC: HVACMode.HEAT,
+    OverkizCommandParam.STANDBY: HVACMode.OFF,
+    OverkizCommandParam.INTERNAL: HVACMode.AUTO,
+}
+
+HVAC_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODE.items()}
+
+
+class AtlanticElectricalHeaterWithAdjustableTemperatureSetpoint(
+    OverkizEntity, ClimateEntity
+):
+    """Representation of Atlantic Electrical Heater (With Adjustable Temperature Setpoint)."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODE_TO_OVERKIZ]
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if OverkizState.CORE_OPERATING_MODE in self.device.states:
+            state = self.device.states[OverkizState.CORE_OPERATING_MODE]
+            return OVERKIZ_TO_HVAC_MODE[OverkizCommandParam(state)]
```

Comment:
I am still not a fan of adding `OverkizCommandParam` here. Why not just pass the string? To overcome the MyPy error, you can type `OVERKIZ_TO_HVAC_MODE`. 

If we want to settle for this new style, we should do this perhaps on all other climate implementations as well :). Would be nice to have them all use the same code style.

Reply:
Sure thing, that's maybe not the prettiest style we can see.

And yes, let's modify the style for all the integration at once.

Reply:
If I removed the call to the constructor, the value is not correctly retrieved from the dict.

```python
    def test_dict_enum(self):
        OVERKIZ_TO_PRESET_MODE = {
            OverkizCommandParam.OFF: "TOTO"
        }
        assert OVERKIZ_TO_PRESET_MODE["OFF"] == "TOTO"
```

> FAILED test_foo - KeyError: 'OFF'